Refactor: Rename extension function orEmpty to orZero for Floats

skaik-mo opened the pull request:
This commit renames the extension function `orEmpty()` to `orZero()` for `Float?` types to better reflect its behavior of returning 0f for null values.

Additionally:
- A new `orEmptyString()` extension function is introduced for `LocalDate?` to return an empty string for null values.
- The `popularity` and `voteAverage` fields in `SeriesDto`, `SeriesDetailsDto`, and `CollectionItemDto` are changed from `Double?` to `Float?` to align with the usage of `Float` for these values elsewhere.
- The `orZero()` extension is now used consistently across mappers for handling nullable Float values related to ratings and popularity.
- The `orEmptyString()` extension is used for formatting release years in mappers.
- A new utility file `extension.kt` is added in `presentation/details` for the `orZero()` extension.

skaik-mo committed:
Refactor: Rename extension function orEmpty to orZero for Floats

This commit renames the extension function `orEmpty()` to `orZero()` for `Float?` types to better reflect its behavior of returning 0f for null values.

Additionally:
- A new `orEmptyString()` extension function is introduced for `LocalDate?` to return an empty string for null values.
- The `popularity` and `voteAverage` fields in `SeriesDto`, `SeriesDetailsDto`, and `CollectionItemDto` are changed from `Double?` to `Float?` to align with the usage of `Float` for these values elsewhere.
- The `orZero()` extension is now used consistently across mappers for handling nullable Float values related to ratings and popularity.
- The `orEmptyString()` extension is used for formatting release years in mappers.
- A new utility file `extension.kt` is added in `presentation/details` for the `orZero()` extension.

mohannadahmed00 commented on presentation/profile/src/main/java/com/giraffe/presentation/profile/utils/Extensions.kt:
```diff
@@ -7,11 +7,12 @@ import com.giraffe.media.exception.NoInternetException
 
 
-fun Any?.orEmpty(): String = this?.toString() ?: ""
+fun LocalDate?.orEmptyString(): String = this?.toString() ?: ""
```
Where is the date formate that should match our design? It's just toString!!!

skaik-mo committed:
Format release date for media items

This commit introduces a new extension function `toFormattedDate()` for `LocalDate?` to display dates in "yyyy, MMM d" format (e.g., "2023, Jan 1").

This new function replaces the previous `orEmptyString()` in the media item mappers to provide a more user-friendly date representation.
